.github/workflows/main.yml

```diff
@@ -0,0 +1,39 @@
+
+name: CI
+
+# Run tests on pull requests and when changes are directly
+# commited to master.
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  # Run static/code-quality checks
+  check:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/setup-go@v2-beta
+      with:
+        go-version: 1.14
+    - uses: actions/checkout@v2
+    - name: Install revive
+      run: go get github.com/mgechev/revive
+    - name: Run checks
+      run: make check
+
+  # Run the test suite in a container per-ceph-codename
+  test-suite:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        ceph_version:
+        - "luminous"
+        - "mimic"
+        - "nautilus"
+        - "octopus"
+    steps:
+    - uses: actions/checkout@v2
+    - name: Run test container
```

Comment:
how do we know the container stuff is installed? I guess I'm missing something...

Reply:
It's kinda buried in the documentation. Worst case scenario it breaks if they change something and then we just have to add a line to install docker or whatnot.